game/unicode.cc

```diff
@@ -12,63 +12,73 @@ UErrorCode UnicodeUtil::m_icuError = U_ZERO_ERROR;
+        match = UnicodeUtil::getCharset(data);
+    }
+    else {
+        match = std::pair<std::string,int>("UTF-8",100); // If there's no filename, assume it's internal text and thus utf-8.
+    }
+    icu::UnicodeString ustring;
+    if (data.substr(0, 3) == "\xEF\xBB\xBF") {
+        if (config["game/bom_warnings"].b()) {
+            std::clog << "unicode/warning: " << _filename << " UTF-8 BOM ignored. Please avoid editors that add a BOM to UTF-8 (e.g. Notepad)." << std::endl;
+        }
+        match.first = "UTF-8";
+        match.second = 100;
+        _stream.str(data.substr(3)); // Remove BOM if there is one
+    }
+    std::clog << "unicode/debug: Detected encoding, " << match.first << ", confidence for encoding is: " << match.second << std::endl;
+    if (match.second > 10 && match.second < 50) { // 50 is a really good match, 10 means an encoding that could be conceivably used to display the text.
+        if (match.first == "ISO-8859-1" || match.first == "ISO-8859-2") {
+            match.first = "UTF-8";
+            match.second = 75;	// Mostly western characters. Let's treat it as a UTF-8 false-negative.
+        }
+    }
+    if (match.second >= 50) { // fairly good match?
+        std::string charset = match.first;
+        if (charset != "UTF-8") {
+            if (!_filename.empty()) { std::clog << "unicode/warning: " << _filename << " is not UTF-8... (" << charset << ") detected. Use a text-editor or other utility to convert your files." << std::endl; }
+            std::string _str;
+            const char* tmp = data.c_str();
+            icu::UnicodeString ustring = icu::UnicodeString(tmp, charset.c_str());
+            _stream.str(ustring.toUTF8String(_str));
+        }
+    }
 }
 
 std::string convertToUTF8(std::string const& str) {
-	std::stringstream ss(str);
-	convertToUTF8(ss, std::string());
-	return ss.str();
+    std::stringstream ss(str);
+    convertToUTF8(ss, std::string());
+    return ss.str();
 }
 
 std::string unicodeCollate(std::string const& str) {
-	ConfigItem::StringList termsToCollate = config["game/sorting_ignore"].sl();
-	std::string pattern = std::string("^((");
-	for (auto term: termsToCollate) {
-		if (term != termsToCollate.front()) { pattern.append(std::string("|")); }
-		pattern.append(term);
-		if (term == termsToCollate.back()) { pattern.append(std::string(")\\s(.+))$")); }
-	}	
-	std::string collated = boost::regex_replace(convertToUTF8(str), boost::regex(pattern), "\\3 \\2");
-	return collated;
+    ConfigItem::StringList termsToCollate = config["game/sorting_ignore"].sl();
+    std::string pattern = std::string("^((");
+    for (auto term: termsToCollate) {
+        if (term != termsToCollate.front()) { pattern.append(std::string("|")); }
+        pattern.append(term);
+        if (term == termsToCollate.back()) { pattern.append(std::string(")\\s(.+))$")); }
+    }	
+    std::string collated = boost::regex_replace(convertToUTF8(str), boost::regex(pattern), "\\3 \\2");
```

Comment:
Instead of `boost::regex_replace` use the c++11 equivalent:

`std::regex_replace(convertToUTF8(str), std::regex(pattern), "\\3 \\2");`

Reply:
This appears to use spaces for indentation. Until now, Performous has been indented by tabs. Is this intentional?

Reply:
@Tronic i'd like to discuss tabs vs spaces in #338